handler/src/main/java/io/netty/handler/ssl/AbstractSniHandler.java

```diff
@@ -80,7 +80,7 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                                 NotSslRecordException e = new NotSslRecordException(
                                         "not an SSL/TLS record: " + ByteBufUtil.hexDump(in));
                                 in.skipBytes(in.readableBytes());
-
+                                ctx.fireUserEventTriggered(new SniCompletionEvent(e));
```

Comment:
A bit on the fence with this one. It's an event that indicates a non-SNI client (if you want to say it that way) or possibly a non-TLS client altogether and could be a event type on its own. On the other hand it's covered by the `hostname` being null further down the road.

Reply:
I think I would fire it like I did here. Just to be more consistent